pkg/chunk/aws/sse_config.go

```diff
@@ -0,0 +1,68 @@
+package aws
+
+import (
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// SSEKMS config type constant to configure S3 server side encryption using KMS
+	// https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html
+	SSEKMS     = "SSE-KMS"
+	sseKMSType = "aws:kms"
+	// SSES3 config type constant to configure S3 server side encryption with AES-256
+	// https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html
+	SSES3     = "SSE-S3"
+	sseS3Type = "AES256"
+)
+
+// SSEEncryptionConfig configures server side encryption (SSE)
+type SSEEncryptionConfig struct {
```

Comment:
Why do we need this when we already have `SSEConfig`?

Reply:
`SSEConfig` is the struct responsible to receive the config from the CLI or a config file. After `SSEConfig` is filled with the values provided by the user, a `SSEEncryptionConfig` struct will be created to be worked inside the application. In the end, `SSEEncryptionConfig` will have different values than `SSEConfig`.